File: star.js
// row = 5
 
// *
// $ *
// $ * $
// * $ * $
// * $ * $ *

// Get user input for number of rows
let row = 6;

// Initialize starting character
let startChar = '*';

for (let i = 1; i <= row; i++) {
  let line = '';
  let currentChar = startChar;

  // Generate pattern for each row
  for (let j = 1; j <= i; j++) {
    line += currentChar + ' ';
    currentChar = (currentChar === '*') ? '$' : '*';
  }

  console.log(line.trim());

  // Decide next line's starting character (based on last printed char)
  startChar = (line.trim().endsWith('*')) ? '$' : '*';
}

File: star.java

public class Star{
    public static void main(String[] args) {
        int row = 5;

        String startChar = "*";

        for (int i = 1; i <= row; i++) {
            String line = "";
            String currentChar = startChar;
            for (int j = 1; j <= i; j++) {
                line += currentChar+" ";
                currentChar = (currentChar.equals("*")) ? "$" : "*";
            }
            System.out.println(line);
            startChar = (line.trim().endsWith("*")) ? "$" : "*";
        }
    }
}